packages/react-native-passkey-stamper/src/index.ts

```diff
@@ -0,0 +1,155 @@
+/// <reference lib="dom" />
+import { Passkey } from "react-native-passkey";
+import { sha256 } from "@noble/hashes/sha256";
+import type { TurnkeyApiTypes } from "@turnkey/http";
+
+// https://www.w3.org/TR/webauthn-2/#dictionary-credential-descriptor
+// Copied from https://github.com/f-23/react-native-passkey/blob/17184a1b1f6f3ac61e07aa784c9b64efb28b570e/src/Passkey.tsx#L80C1-L85C2
+// TODO: can we import this type instead?
+interface PublicKeyCredentialDescriptor {
+  type: string;
+  id: string;
+  transports?: Array<string>;
+}
+
+export type TurnkeyAuthenticatorParams =
+  TurnkeyApiTypes["v1AuthenticatorParamsV2"];
+
+// Header name for a webauthn stamp
+const stampHeaderName = "X-Stamp-Webauthn";
+
+export type TPasskeyRegistrationConfig = {
+  // The RPID ("Relying Party ID") for your app.
+  // See https://github.com/f-23/react-native-passkey?tab=readme-ov-file#configuration to set this up.
+  rpId: string;
+  rpName: string;
+  // Properties for passkey display: user name and email will show up in the prompts
+  user: {
+    id: string;
+    name: string;
+    displayName: string;
+  };
+  // Name of the authenticator (affects Turnkey only, won't be shown on passkey prompts)
+  // TODO: document restrictions on character sets
+  authenticatorName: string;
+  // Optional challenge. If not provided, a new random challenge will be generated
+  challenge?: string;
+  // Optional timeout value. Defaults to 5 minutes.
+  timeout?: number;
+  // Optional override for UV flag. Defaults to "preferred".
+  userVerification?: UserVerificationRequirement;
+  // Optional list of credentials to pass. Defaults to empty
+  allowCredentials?: PublicKeyCredentialDescriptor[];
+};
+
+export type TPasskeyStamperConfig = {
+  // The RPID ("Relying Party ID") for your app.
+  // See https://github.com/f-23/react-native-passkey?tab=readme-ov-file#configuration to set this up.
+  rpId: string;
+  // Optional timeout value. Defaults to 5 minutes.
+  timeout?: number;
+  // Optional override for UV flag. Defaults to "preferred".
+  userVerification?: UserVerificationRequirement;
+  // Optional list of credentials to pass. Defaults to empty
+  allowCredentials?: PublicKeyCredentialDescriptor[];
+};
+
+const defaultTimeout = 5 * 60 * 1000; // five minutes
+const defaultUserVerification = "preferred";
+
+/**
+ * Creates a passkey and returns authenticator params
+ */
+export async function CreatePasskey(
+  config: TPasskeyRegistrationConfig
+): Promise<TurnkeyAuthenticatorParams> {
```

Comment:
`TurnkeyAuthenticatorParams`  contains a lot more fields than is used in this function. For example:

  // Optional challenge. If not provided, a new random challenge will be generated
  challenge?: string;
  
I cannot see that we pass along `challenge` from config to `Passkey.register({`.

Only thing used seems to be `config.user`, `config. rpId `, `config. rpName `?

Maybe we can use similar interface as [this](https://github.com/f-23/react-native-passkey/blob/17184a1b1f6f3ac61e07aa784c9b64efb28b570e/src/__tests__/testData/RegRequest.json), set our defaults and spread the rest? So we would need to change rpName and rpId to be `rp` object. 

Something like:

```
const {challenge, ...rest} = config;

  const registrationResult = await Passkey.register({
    challenge: challenge || getRandomChallenge(),
    ...rest,
    pubKeyCredParams: pubKeyCredParams || [
      {
        type: "public-key",
        alg: -7,
      },
      {
        type: "public-key",
        alg: -257,
      },
    ],
  });
  
```

Reply:
That's a good catch, I forgot to make the challenge generation optional!

For pubKeyCredParams I actually think this shouldn't be overridable because these are the only algorithms Turnkey supports right now.

> `TurnkeyAuthenticatorParams` contains a lot more fields than is used in this function

This is the return type of this function. Do you mean `PasskeyRegistrationRequest` instead? (if so, agreed: things like excludeCredential wouldn't be overridable). Also, I should probably take an extra options param so that `withSecurityKey` can be set!

Reply:
Oh, i forgot removing that. `pubKeyCredParams` shouldnt be optional (and it is not in the ts type either so dont know why i added it :P).

Reply:
Addressed in https://github.com/tkhq/sdk/pull/209/commits/b98d1f43641afa9016577c9be555731e539cef70, let me know what you think! I went with the "explicit" approach so that each param can be documented and tweaked if needed.